### DISCUSSION TITLE
The size of tensor Runtime Error

### DISCUSSION THREAD

**Discussion by umichwbaek** [mir-group/allegro#33] [Q&A]
Hi

I got an Runtime error while running the nequip-train of my work.

File "/home/software/anaconda3/envs/NequiP/lib/python3.10/site-packages/nequip/train/metrics.py", line 183, in __call__                                                                                                                     error_N = error / N
RuntimeError: The size of tensor a (350) must match the size of tensor b (5) at non-singleton dimension 0

I found that the error comes from the metrics_components. When I mute metrics_components, it runs without the errors, but I want to see the output with components I put.

Are there have any additional parameters that I need to set?

**Comment by Linux-cpp-lisp** [comment]
Hi @umichwbaek ,

Can you please provide more details about your config and dataset?

**Comment by umichwbaek** [comment]
I used almost the same input parameters of example.yaml in allegro except for the following:

r_max: 7.0

dataset: ase
ase_args:
    format: extxyz
key_mapping:
  z: atomic_numbers
  E: total_energy
  F: forces
  R: pos

validation_dataset: ase

n_train: 160
n_val: 1044

batch_size: 2
max_epochs: 1000

metrics_components:
  - - total_energy
    - mae
    - PerAtom: True
  - - total_energy
    - rmse
    - PerAtom: True
  - - forces
    - mae
    - PerAtom: True
  - - forces
    - rmse
    - PerAtom: True
   

The type of train and validation dataset is xyz. I have separate validation dataset file, so I did not use 'train_val_split'. It did not show any dataset error, so the format would not be the problem

**Comment by Linux-cpp-lisp** [comment]
Ah I see it--- you should not set `PerAtom: True` for metrics on per-atom quantities like forces. Perhaps that is a little unclear, but it specifically is only for getting the version of a _global_ quantity normalized by the number of atoms. I'll try to add a clearer error.

**Comment by umichwbaek** [comment]
Oh I see. Then how can I get per-atom values of mse and rmse? Is it the mse or rmse divided by the total number of atoms of the sum of whole data, or average?

**Comment by Linux-cpp-lisp** [comment]
You mean you want to report the force RMSE on a specific atom?

**Comment by umichwbaek** [comment]
Both. For overall and specific if possible.

**Reply by Linux-cpp-lisp** [reply]
What do you mean by "For overall and specific if possible."?

**Reply by umichwbaek** [reply]
I want to know 'rmse/the number of all atom' and if possible 'rmse/individual atoms'.

**Reply by Linux-cpp-lisp** [reply]
Is that supposed to be division in the first case and not in the second?

If you want to do some post processing on per-atom level errors, see https://github.com/mir-group/nequip/blob/develop/examples/parity_plot.py for an example. If you want to track the error of one specific atom during training, that will take a little bit of coding (though not much). If you want per-species errors, see https://github.com/mir-group/nequip/blob/main/configs/full.yaml#LL259C31-L259C31
